src/main/java/com/minecolonies/rankup/Rankup.java

```diff
@@ -185,20 +199,22 @@ public void reload() throws IOException
         this.container.reloadSystemConfig();
         reloadables.values().forEach(Action::action);
 
-        this.getAllConfigs().remove(GroupsConfig.class);
         this.getAllConfigs().remove(AccountConfigData.class);
 
         final Path accountsPath = this.getConfigDir().resolve("playerstats.conf");
         AccountConfigData data = this.getConfig(accountsPath, AccountConfigData.class,
           HoconConfigurationLoader.builder().setPath(accountsPath).build());
         this.getAllConfigs().put(AccountConfigData.class, data);
 
-        final Path groupsPath = this.getConfigDir().resolve("groups.conf");
-        GroupsConfig groups = this.getConfig(groupsPath, GroupsConfig.class,
-          HoconConfigurationLoader.builder().setPath(groupsPath).build());
-        this.getAllConfigs().put(GroupsConfig.class, groups);
+        CoreConfig config = getConfigAdapter(CoreModule.ID, CoreConfigAdapter.class).get().getNodeOrDefault();
 
-        generateGroups();
+        for (final String name : config.groupConfigs)
+        {
+            final Path confPath = getConfigDir().resolve("group-configs").resolve(name);
+            GroupsConfig groups = getConfig(confPath, GroupsConfig.class,
+              HoconConfigurationLoader.builder().setPath(confPath).build());
+            getGroupConfigs().add(groups);
+        }
     }
 
     public static Rankup getInstance() { return instance; }
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this line start at column 9. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AIndentationCheck)